modules/web/client/src/main/scala/observe/ui/components/MainApp.scala

```diff
@@ -243,13 +270,19 @@ object MainApp:
                 IO.println(t.getMessage) >>
                 isSynced.async.set(SyncStatus.OutOfSync) // Triggers reSync
           )
-      .useResourceOnMount:
-        // Reconnect(WebSocketClient[IO]).connectHighLevel(WSRequest(EventWsUri))
-        // We also have to reSync in case of connection lost
-        WebSocketClient[IO].connectHighLevel(WSRequest(EventWsUri))
+      // Connection to event stream is surrogated to ODB WS connection,
+      // only established whenever ODB WS is connected and initialized.
+      .useStreamBy((_, _, _, _, ctxPot, _, _, _) => ctxPot.void): (_, _, _, _, ctxPot, _, _, _) =>
+        _ => ctxPot.map(_.odbClient).toOption.foldMap(_.statusStream)
+      .useResourceBy((_, _, _, _, _, _, _, _, odbStatus) => odbStatus):
+        (_, toastRef, isSynced, _, ctxPot, rootModelDataPot, environmentPot, _, _) =>
+          case PotOption.ReadySome(PersistentClientStatus.Initialized) =>
+            // Reconnect(WebSocketClient[IO]).connectHighLevel(WSRequest(EventWsUri))
```

Comment:
Commented code

Reply:
Yeah that is still waiting for the reconnection middleware to be ready in http4s.